Updated version of `setuptools` rejects the `version.txt` ("dev") from `master`

The following error is currently affecting open PRs:

```
Installing spinalcordtoolbox...

Obtaining file:///home/runner/work/spinalcordtoolbox/spinalcordtoolbox
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      running egg_info
      /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/dist.py:543: UserWarning: The version specified ('dev') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
        warnings.warn(
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/setup.py", line 16, in <module>
          setup(
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/dist.py", line 1208, in run_command
          super().run_command(command)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.ensure_finalized()
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 111, in ensure_finalized
          self.finalize_options()
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 219, in finalize_options
          parsed_version = parse_version(self.egg_version)
        File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/pkg_resources/_vendor/packaging/version.py", line 266, in __init__
          raise InvalidVersion(f"Invalid version: '{version}'")
      pkg_resources.extern.packaging.version.InvalidVersion: Invalid version: 'dev'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

Failed running pip install: 1


Installation failed!

Please find the file "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/install_sct_log.txt",
then upload it as a .txt attachment in a new topic on SCT's forum:
--> http://forum.spinalcordmri.org/c/sct

Error: Process completed with exit code 1.
```

Our `version.txt` on master is the following:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/1b5c49c3904dffe63546adbadc47759a53fbc8ca/spinalcordtoolbox/version.txt#L1

Investigation to come -- just quickly opening up an issue.

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -4,10 +4,10 @@ on:
   workflow_dispatch:
     inputs:
       milestone_title:
-        description: 'Milestone title'
+        description: 'Milestone title (this release)'
         required: true
       future_milestone_title:
-        description: 'Milestone title (future release)'
+        description: 'Milestone title (next release)'
         required: true
   push:
     branches:

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -149,16 +149,16 @@ jobs:
           git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
           git push
 
-    - name: Download requirements-freeze.txt
+    - name: Download requirements-freeze.txt (for release)
       uses: actions/download-artifact@v3
       with:
         name: requirements-freeze
 
-    - name: Update version.txt (using milestone title)
+    - name: Update version.txt (for release)
       run: |
         echo "${{ github.event.inputs.milestone_title }}" > spinalcordtoolbox/version.txt
 
-    - name: Commit and push
+    - name: Commit and push release changes
       run: |
         git checkout -b bot/${{ github.event.inputs.milestone_title }}
         git add requirements-freeze.txt

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -128,6 +128,11 @@ jobs:
     # Only create the release if workflow is run manually. (This allows the other steps in the workflow to test PRs.)
     if: github.event_name == 'workflow_dispatch'
     steps:
+    # The GitHub Actions bot email was taken from: https://github.community/t/github-actions-bot-email-address/17204/6
+    - name: Set bot user data for commits
+      run: |
+        git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git config --global user.name "GitHub Actions Bot"
 
     - name: Checkout spinalcordtoolbox (main branch)
       uses: actions/checkout@v2
@@ -143,12 +148,6 @@ jobs:
     - name: Update version.txt (using milestone title)
       run: |
         echo "${{ github.event.inputs.milestone_title }}" > spinalcordtoolbox/version.txt
-
-    # The GitHub Actions bot email was taken from: https://github.community/t/github-actions-bot-email-address/17204/6
-    - name: Set bot user data for commits
-      run: |
-        git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
-        git config --global user.name "GitHub Actions Bot"
 
     - name: Commit and push
       run: |

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -6,6 +6,9 @@ on:
       milestone_title:
         description: 'Milestone title'
         required: true
+      future_milestone_title:
+        description: 'Milestone title (future release)'
+        required: true
   push:
     branches:
       - master
@@ -139,6 +142,12 @@ jobs:
       with:
         ref: ${{ env.MAIN_BRANCH }}
         fetch-depth: 0
+
+    - name: Update version.txt (main branch)
+      run: | 
+          echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
+          git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
+          git push
 
     - name: Download requirements-freeze.txt
       uses: actions/download-artifact@v3

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -143,6 +143,11 @@ jobs:
         ref: ${{ env.MAIN_BRANCH }}
         fetch-depth: 0
 
+    # Branch off before adding commits to the main branch
+    - name: Create release branch
+      run: |
+        git branch bot/${{ github.event.inputs.milestone_title }}
+
     - name: Update version.txt (main branch)
       run: |
         echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
@@ -161,7 +166,7 @@ jobs:
 
     - name: Commit and push release changes
       run: |
-        git checkout -b bot/${{ github.event.inputs.milestone_title }}
+        git checkout bot/${{ github.event.inputs.milestone_title }}
         git add requirements-freeze.txt
         git commit -m "Update requirements-freeze.txt for ${{ github.event.inputs.milestone_title }}"
         git add spinalcordtoolbox/version.txt

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -160,17 +160,19 @@ jobs:
       with:
         name: requirements-freeze
 
-    - name: Update version.txt (for release)
-      run: |
-        echo "${{ github.event.inputs.milestone_title }}" > spinalcordtoolbox/version.txt
-
-    - name: Commit and push release changes
-      run: |
+    - name: Commit requirements-freeze.txt (for release)
         git checkout bot/${{ github.event.inputs.milestone_title }}
         git add requirements-freeze.txt
         git commit -m "Update requirements-freeze.txt for ${{ github.event.inputs.milestone_title }}"
+
+    - name: Update version.txt (for release)
+      run: |
+        echo "${{ github.event.inputs.milestone_title }}" > spinalcordtoolbox/version.txt
         git add spinalcordtoolbox/version.txt
         git commit -m "Update version.txt for ${{ github.event.inputs.milestone_title }}"
+
+    - name: Push release changes
+      run: |
         git tag ${{ github.event.inputs.milestone_title }}
         git push --tags
 

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -144,11 +144,11 @@ jobs:
         fetch-depth: 0
 
     - name: Update version.txt (main branch)
-      run: | 
-          echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
-          git add spinalcordtoolbox/version.txt
-          git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
-          git push
+      run: |
+        echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
+        git add spinalcordtoolbox/version.txt
+        git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
+        git push
 
     - name: Download requirements-freeze.txt (for release)
       uses: actions/download-artifact@v3

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -161,6 +161,7 @@ jobs:
         name: requirements-freeze
 
     - name: Commit requirements-freeze.txt (for release)
+      run: |
         git checkout bot/${{ github.event.inputs.milestone_title }}
         git add requirements-freeze.txt
         git commit -m "Update requirements-freeze.txt for ${{ github.event.inputs.milestone_title }}"

--- a/.github/workflows/create-release.yml
+++ b/.github/workflows/create-release.yml
@@ -146,6 +146,7 @@ jobs:
     - name: Update version.txt (main branch)
       run: | 
           echo "${{ github.event.inputs.future_milestone_title }}.dev0" > spinalcordtoolbox/version.txt
+          git add spinalcordtoolbox/version.txt
           git commit -m "Update version.txt for ${{ github.event.inputs.future_milestone_title }} development"
           git push
 

